Update UG (#90)
AY2223S1-CS2103T-W17-2/tp: docs/UserGuide.md

   open the help window.<br>
   Some example commands you can try:

> 

Reviewer: Should we add a info block here to let users know of the notations/syntax we use? Such as code blocks etc.

Author: i think these were mentioned earlier command format portion, but we can add another code block for each sample commands to make it clearer!

>     - **`list`** : Lists all contacts.

Reviewer 2: I think dh this anymore

Author: removed

>     - **`add`**`t/i d/TutionTeaching a/45.00 da/13-10-2022 c/Salary` : Adds an income entry

Reviewer 2: Teaching Tuition instead ?

Author: Resolved

      with the description `TuitionTeaching` to the PennyWise application.

>     - **`delete`**`3 t/e` : Deletes the 3rd contact shown in the expenditure.

Reviewer 2: Deletes the 3rd entry in the expenditure list

**:information_source: Notes about the command format:**<br>

- All command follow this format:<br>
  `command | entry identifier | input | [optional inputs]`<br>
  Example: `add t/e d/Lunch a/15.60 da/15-08-2022 c/Meal` <br>

>     - Command : add<br>

Reviewer 3: Same for the Identifier and Input

>     - Command : add<br>

Author: Have placed these in code blocks

</div>

## Before Using

<div markdown="block" class="alert alert-info">

**:information_source: Notes about the application:**<br>
- For **ALL** entries, categories are <ins>COMPULSORY</ins> and every entry can only contain <ins>one</ins> category. 
  The following table shows the`Expenditure` and `Income` entries categories. The category names are <ins>case sensitive</ins>.

|   `Expenditure`   |    `Income`     |
|:-----------------:|:---------------:|
|     **Food**      |   **Salary**    |
|   **Groceries**   |  **Allowance**  |
| **Entertainment** |   **Profit**    |
|   **Education**   | **Investments** |
|    **Housing**    |    **Gifts**    |
|    **Others**     |   **Others**    |

- For **ALL** entries, 2 entries are considered duplicates <ins>IF</ins> both entries have exactly the same:
  `description`, `date` `amount` and `category`. We do not want PennyWise to be managing entries which are a repeat,
  of one another, as it would be easier to simply use the [edit](#editing-entries--edit-index_of_entry-tentry_type-dedited_description-aedited_amount-daedited_date-cedited_category) 
  command to alter the original entry's specifications.

>   e.g. Executing the following 2 commands one after another, PennyWise will recognise (2) as a duplicate entry. 

Reviewer: ```suggestion
  For example, when executing the following 2 commands one after another, PennyWise will recognise (2), which is the second `Teh Beng` added as a duplicate entry. 
```

  <pre>
    <code>

>         1. add t/e d/Teh Beng a/1.50 da/22-10-2022 c/Food

Reviewer 3: Could you also add in the rationale behind this? That having multiple duplicate entries will make it difficult for users to track their expenses later on and will make their entry list messy. Having it differentiated will make it easier for users to recall what they spent their money on. 

You can also mention that they can update the description and name as such `2 Teh Beng` and `3.00`. But it's just a suggestion and not necessary

Author: have changed!

        2. add t/e d/Teh Beng a/1.50 da/22-10-2022 c/Food
    </code>
  </pre>
  To add similar entries, vary the description/amount/date/category to let PennyWise know they are not duplicates!
  <pre>
    <code>
        1. add t/e d/Teh Beng 11am a/1.50 da/22-10-2022 c/Food
        2. add t/e d/Teh Beng 4pm a/1.50 da/22-10-2022 c/Food 
    </code>
  </pre>

</div>

### Adding entries : `add t/ENTRY_TYPE d/DESCRIPTION a/AMOUNT da/DATE c/CATEGORY`
Adds an entry to the specified list. <br>
1. Add expense <br>
   - Example: `add t/e d/Lunch a/15.60 da/10-10-2022 c/Food`<br>

>    - An expense of 15.60 on 10/10/2022 will be added to the Expenditure list, under the Food category.

Reviewer: ```suggestion
   - An expense of $15.60 on 10/10/2022 will be added to the Expenditure list, under the Food category.
```

Might it be better to use dollar signs when explaining?

1. Add income <br>
   - Example: `add t/i d/Tuition a/40.00 da/10-10-2022 c/Salary`<br>
   - An income of 40.00 on 10/10/2022 will be added to the Income list, under the Salary category.

### Deleting entries : `del INDEX_OF_ENTRY t/ENTRY_TYPE`
Deletes an entry.
1. Example: `del 2 t/e` deletes the 2nd item on the expenditure list 

>    Expenditure list:

Reviewer: Do you think it's better to supply screenshots? I can help with the screenshots if needed, perhaps after merging this PR in

Author: yeap once the PR is merged in lets add screenshots

   1. Movie 15.60 12 Sep 2022 c/Entertainment
   2. Drink 1.20 12 Sep 2022 c/Food 

>    Expected: `Deleted Entry: Drink; Date: 12-09-2022; Amount: 1.20; Tag: Food`

Reviewer 3: When I viewed the diff the Expected line appears on the same line as the "2. `Drink 1.20  12 Sep 2022 c/Food` line". Not sure if its just GitHub or a bug

Author: have added a `<br>` in between all the expected outcomes to fix this

   
1. Example: `del 2 t/i` deletes the 2nd item on the income list
   Income list:
   1. Tuition 40.00 12 Sep 2022 c/Salary
   2. Allowance 100.00 12 Sep 2022 c/Allowance
   Expected: `Deleted Entry: Allowance; Date: 12-09-2022; Amount: 100.00 Tag: Allowance`

### Editing entries : `edit INDEX_OF_ENTRY t/ENTRY_TYPE [d/EDITED_DESCRIPTION a/EDITED_AMOUNT da/EDITED_DATE c/EDITED_CATEGORY]`
Edits an entry, at least **1** of the optional fields must be present.

1. Edits expenditure with specified fields
   * Example: `edit 1 t/e d/ChickenRice`
   * The expenditure at index 1 will have its description edited
   * Expected: `Edited Entry: ChickenRice; Date: 21-10-2022; Amount: 4.20; Tag: Food`
1. Edits income with specified fields
   * Example: `edit 2 t/i a/150.00 da/22-10-2022`
   * The income at index 2 will have its amount and date edited
   * Expected: `Edited Entry: CafeSalary; Date: 22-10-2022; Amount: 150.00; Tag: Salary`

<div markdown="span" class="alert alert-primary">:bulb: **Tip:**
The default date is the current date on your computer!
</div>

### Summary of entries : `summary [mo/MONTH]`

> 1. Summarise of all entries in the currently shown list

Reviewer: ```suggestion
1. Summary of all entries in the currently shown list
```

   * Examples: `summary`
     * Expected: <br/>
       Financials Summarized <br/>
       Total Expenditure: 154.40 <br/>
       Total Income: 150.00 <br/>
       Total Balance: -4.40 <br/>

> 2. Summarise of all entries the specified month

Reviewer: ```suggestion
2. Summary of all entries the specified month
```

    * Examples: `summary mo/2022-09`
    * Expected: <br/>
      Financials Summarized <br/>
      Total Expenditure: 80.40 <br/>
      Total Income: 50.00 <br/>
      Total Balance: -30.40 <br/>
* Provides a financial summary recorded by the user in a month. The month refers to the month that is displayed to the user.
* The `MONTH` field is optional, if no month is specified, the application displays the summary for all entries.

> ### View entries by category : `view t/ENTRY_TYPE g/GRAPH_TYPE`

Reviewer: ```suggestion
### View entries by category : `view t/ENTRY_TYPE g/GRAPH_TYPE [mo/month]`
```

1. View a PieChart of all expenditures by categories
    * Examples: `view t/e g/c`
    * Expected: `Show graphically all expenditure by category` and a PieChart on the right of the application
1. View a PieChart of all incomes by categories
   * Examples: `view t/i g/c`
   * Expected: `Show graphically all income by category` and a PieChart on the right of the application

> ### View entries by month : `view t/ENTRY_TYPE g/GRAPH_TYPE [mo/MONTH]`

Reviewer: ```suggestion
### View entries by month : `view t/ENTRY_TYPE g/GRAPH_TYPE mo/MONTH`
```

I think `month` should be compulsory here

**:information_source: Notes about the command format:**<br>

> - All command follow this format:<br>

Reviewer 2: might be good to mention that all identifiers are case sensitive

Author: Have added this in

</div>

## Before Using

<div markdown="block" class="alert alert-info">

**:information_source: Notes about the application:**<br>
- For **ALL** entries, categories are <ins>compulsory</ins> and every entry can only contain <ins>one</ins> category. 
  The following table shows the`Expenditure` and `Income` entries categories.

|   `Expenditure`   |    `Income`     |
|:-----------------:|:---------------:|
|     **Food**      |   **Salary**    |
|   **Groceries**   |  **Allowance**  |
| **Entertainment** |   **Profit**    |
|   **Education**   | **Investments** |
|    **Housing**    |    **Gifts**    |
|    **Others**     |   **Others**    |

- For **ALL** entries, 2 entries are considered duplicates <ins>IF</ins> both entries have exactly the same:
  `description`, `date` `amount` and `category`. We do not want PennyWise to be managing entries which are a repeat,
  of one another, as it would be easier to simply use the [edit](#editing-entries--edit-index_of_entry-tentry_type-dedited_description-aedited_amount-daedited_date-cedited_category) 
  command to alter the original entry's specifications.

>   e.g. Executing the following 2 commands one after another, PennyWise will recognise (2) as a duplicate entry. 

Reviewer 2: would adding a second portion on what is accepted be gd? like maybe varying description or amount

   Expected: `Deleted Entry: Drink; Date: 12-09-2022; Amount: 1.20; Tag: Food`
   
1. Example: `del 2 t/i` deletes the 2nd item on the income list
   Income list:
   1. Tuition 40.00 12 Sep 2022 c/Salary
   2. Allowance 100.00 12 Sep 2022 c/Allowance
   Expected: `Deleted Entry: Allowance; Date: 12-09-2022; Amount: 100.00 Tag: Allowance`

### Editing entries : `edit INDEX_OF_ENTRY t/ENTRY_TYPE [d/EDITED_DESCRIPTION a/EDITED_AMOUNT da/EDITED_DATE c/EDITED_CATEGORY]`
Edits an entry, at least **1** of the optional fields must be present.

1. Edits expenditure with specified fields
   * Example: `edit 1 t/e d/ChickenRice`
   * The expenditure at index 1 will have its description edited
   * Expected: `Edited Entry: ChickenRice; Date: 21-10-2022; Amount: 4.20; Tag: Food`
1. Edits income with specified fields
   * Example: `edit 2 t/i a/150.00 da/22-10-2022`
   * The income at index 2 will have its amount and date edited
   * Expected: `Edited Entry: CafeSalary; Date: 22-10-2022; Amount: 150.00; Tag: Salary`

<div markdown="span" class="alert alert-primary">:bulb: **Tip:**
The default date is the current date on your computer!
</div>

### Summary of entries : `summary [mo/MONTH]`

> 1. View a summary of all entires

Reviewer 2: should we add that its the current shown list?

Author: Have added this as well

> 1. View a summary of all entires

Reviewer 2: Should we change word view to avoid confusion w view command? maybe "Summarise all entries" and "Summarise all entries in specified month"

Author: have changed it to **Summarise all entires...** instead

   * Examples: `summary`
     * Expected: <br/>
       Financials Summarized <br/>
       Total Expenditure: 154.40 <br/>
       Total Income: 150.00 <br/>
       Total Balance: -4.40 <br/>
2. View a summary of all entires the specified month
    * Examples: `summary mo/2022-09`

>     * Expected: A summary of entries in September 2022: `Total amount spent in September 2022: $37.70`

Author: have fixed this

* Provides a financial summary recorded by the user in a month. The month refers to the month that is displayed to the user.

> * The `MONTH` field is optional, if no month is specified, the application displays the current month.

Reviewer 2: I think this might be confusing as current if no month is specified all the entries is summarised

Author: have changed this to:
if no month is specified, the application displays the summary for all entries.

the data of your previous PennyWise home folder.

---

## Command summary

| Action              | Format, Examples                                                                                                                                        |
|---------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------|
| **Add**             | `add t/ENTRY_TYPE d/DESCRIPTION a/AMOUNT da/DATE c/CATEGORY` <br> e.g., `add t/e d/Lunch a/15.60 da/10-10-2022 c/Food`                                  |
| **Delete**          | `del INDEX_OF_ENTRY t/ENTRY_TYPE` <br> e.g., `del 2 t/e`                                                                                                |
| **Edit**            | `edit INDEX_OF_ENTRY t/ENTRY_TYPE [d/EDITED_DESCRIPTION a/EDITED_AMOUNT da/EDITED_DATE c/EDITED_CATEGORY]`<br> e.g.,`edit 2 t/i a/150.00 da/22-10-2022` |
| **Summary**         | `summary [mo/MONTH]`<br> e.g., `summary mo/2022-09`                                                                                                     |

> | **View (Category)** | `view t/ENTRY_TYPE g/GRAPH_TYPE` <br> e.g,, `view t/e g/c`                                                                                              |

Reviewer 3: I think it is not necessary to follow e.g. with a comma, just directly e.g. `command` instead. Over here, there's a typo of e.g,,

Author: resolved!
